gwtlib/worktrees.py

```
@@ -169,58 +293,345 @@ def remove_worktree(branch_name: str, git_dir: str) -> None:
+                else:
+                    raise RuntimeError(f"Failed to delete remote branch: {error_msg}")
+
+        # Step 2: Delete local branch (skip if doesn't exist)
+        if branch_exists_locally(branch_name, git_dir):
+            try:
+                run_git_command(["branch", "-D", branch_name], git_dir, capture=False)
+                print(f"Deleted local branch '{branch_name}'", file=sys.stderr)
+                deleted_local = True
+            except subprocess.CalledProcessError as e:
+                error_msg = e.stderr.strip() if e.stderr else str(e)
+                raise RuntimeError(f"Failed to delete local branch: {error_msg}")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)